### PR TITLE
Cb/sim stage updates

### PR DESCRIPTION
**Pull request by claytonpbarrows**
- [x] update stage validation
 - [x] update stage construction
 - [x] update stage update forecast retrieval
 - [x] update `UpdateRef` to store the label accessors
 - [ ] update and enable simulation tests

### CODE REVIEW

File: src/core/core_structs/simulations_stages.jl [hunk 1]
```diff
@@ -60,6 +60,9 @@ end
 mutable struct Stage <: AbstractStage
     op_problem::Type{<:AbstractOperationsProblem}
     model::OperationsTemplate
+    initial_time::Dates.DateTime
+    horizon::Int64
```
Comment (jd-lara) [inline src/core/core_structs/simulations_stages.jl]: This information will be duplicated in canonical. I am fine with that, just warning about it.

File: src/simulation/stage_update.jl [hunk 2]
```diff
@@ -6,12 +6,15 @@ function parameter_update!(param_reference::UpdateRef{T},
 
     devices = PSY.get_components(T, sim.stages[stage_number].sys)
     initial_forecast_time = sim.ref.date_ref[stage_number]
```
Comment (jd-lara) [inline src/simulation/stage_update.jl]: Can we add getter functions for this? I think it will be better.

**Comment by claytonpbarrows** [comment]
An example is here: https://github.com/NREL-SIIP/Examples/blob/tutorials-refactor/notebook/PowerSimulations_examples/sequential_simulations.ipynb

**Comment by claytonpbarrows** [comment]
This is working. @jd-lara an RTS example is here: https://github.com/NREL-SIIP/Examples/blob/tutorials-refactor/script/PowerSimulations_examples/sequential_simulations.jl

**Pull request merged**